datumaro/components/explorer.py

```diff
@@ -43,22 +51,29 @@ def __init__(
         """
         self._model = ExplorerLauncher(model_name="clip_visual_ViT-B_32")
         self._text_model = ExplorerLauncher(model_name="clip_text_ViT-B_32")
-        inference = dataset.run_model(self._model, append_annotation=True)
         self._topk = topk
-
         database_keys = []
         item_list = []
 
-        for item in inference:
-            for annotation in item.annotations:
-                if isinstance(annotation, HashKey):
-                    try:
-                        hash_key = annotation.hash_key[0]
-                        hash_key = np.unpackbits(hash_key, axis=-1)
-                        database_keys.append(hash_key)
-                        item_list.append(item)
-                    except Exception:
-                        hash_key = None
+        datasets = datasets[0]
+        if isinstance(datasets, IDataset):
+            datasets = [datasets]
+
+        for dataset in datasets:
+            uninferenced_dataset = select_uninferenced_dataset(dataset)
+            uninferenced_dataset.run_model(self._model, append_annotation=True)
+            dataset.update(uninferenced_dataset)
+
+            for item in dataset:
+                for annotation in item.annotations:
+                    if isinstance(annotation, HashKey):
+                        try:
+                            hash_key = annotation.hash_key[0]
+                            hash_key = np.unpackbits(hash_key, axis=-1)
+                            database_keys.append(hash_key)
+                            item_list.append(item)
+                        except Exception:
+                            hash_key = None
```

Comment:
```suggestion
                            continue
```
Do we have to set `hash_key = None` explicitly?